Update actions/upload-artifact to v3.1.1

mattgauntseo-sentry opened the pull request:
Committed via https://github.com/asottile/all-repos

evanh reviewed (approved):
Is there any context to these PRs? Or are they auto-generated? It would be nice to get a little context on why these changes are necessary.

mattgauntseo-sentry commented:
Apologies for lack of context, ideally I'd add it to the PR Description (https://github.com/asottile/all-repos/issues/259).

GitHub is planning two large deprecations that affect a broad range of our actions.

https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/
https://github.blog/changelog/2022-09-22-github-actions-all-actions-will-begin-running-on-node16-instead-of-node12/

This version of the upload-artifact addresses these issues.
